File: apps/api/internal/model/bulk.go
package model

type BulkError struct {
	RowNumber int    `json:"rowNumber"`
	Message   string `json:"message"`
}

type BulkResult struct {
	TotalCount     int           `json:"totalCount"`
	SuccessCount   int           `json:"successCount"`
	ErrorCount     int           `json:"errorCount"`
	SuccessfulRows []interface{} `json:"successfulRows"`
	Errors         []BulkError   `json:"errors"`
}

File: apps/web/src/types/bulk.ts
/**
 * Shared CSV bulk-import result (spec/api.md#Bulk Import Users /
 * #Bulk Import Companies, apps/api/internal/model/bulk.go).
 */

export type BulkError = {
  rowNumber: number
  message: string
}

export type BulkResult<T> = {
  totalCount: number
  successCount: number
  errorCount: number
  successfulRows: T[]
  errors: BulkError[]
}
